packages/component/src/hooks/useIntervalSince.js

```diff
@@ -0,0 +1,25 @@
+import { useCallback, useState } from 'react';
+
+import useTimer from './useTimer';
+
+// The `nextTimer` function calculates the next absolute time that the timer should be fired based on the origin (original time received), interval, and current time.
+// If the origin is t=260, and we are currently at t=1000, nextTimer must return t=60260.
+// If the origin is t=260, and we are currently at t=60260 (exact landing), we must return t=120260, not t=60260.
+// This is for fixing bug #2103: https://github.com/microsoft/BotFramework-WebChat/issues/2103.
+
+function nextTimer(origin, interval) {
+  const time = new Date(origin).getTime();
+  const now = Date.now();
+
+  return time > now ? time : now + interval - ((now - time) % interval);
+}
+
+export default function useIntervalSince(origin, interval, fn) {
+  const [timer, setTimer] = useState(nextTimer(origin, interval));
```

Comment:
Can you use `useRef` here?

So if no `fn` is passed in, this will not causing any render.

Reply:
If we use `useRef`, it won't trigger the `useEffect` hook in `useTimer` since the ref isn't updated until after the component is rerendered.

Reply:
I think this question is related to the one above. So I am resolving this and continue to discuss above.